src/gentropy/dataset/dataset.py

```diff
@@ -171,6 +190,46 @@ def validate_schema(self: Dataset) -> None:
                 f"The following fields present differences in their datatypes: {fields_with_different_observed_datatype}."
             )
 
+    def valid_rows(self: Self, invalid_flags: list[str], invalid: bool = False) -> Self:
+        """Filters `Dataset` according to a list of quality control flags. Only `Dataset` classes with a QC column can be validated.
+
+        Args:
+            invalid_flags (list[str]): List of quality control flags to be excluded.
+            invalid (bool): If True returns the invalid rows, instead of the valids. Defaults to False.
+
+        Returns:
+            Self: filtered dataset.
+
+        Raises:
+            ValueError: If the Dataset does not contain a QC column.
+        """
+        # If the invalid flags are not valid quality checks (enum) for this Dataset we raise an error:
+        for flag in invalid_flags:
+            if flag not in self.get_QC_categories():
+                raise ValueError(
+                    f"{flag} is not a valid QC flag for {type(self).__name__}."
+                )
```

Comment:
It's very subjective, and really just my feeling, but I think there's no need to validate the flags here because they represent exclusive criteria. Like, OK we are not including records with `XXX` flags, doesn't matter if `XXX` are valid flags or not. This validation can happen somewhere, but not sure about its right place. Don't get me wrong, this is absolutely fine.

Reply:
Ok, I think I know what my problem is: this logic assumes that the `StudyLocusQualityCheck` class has all the flags. However, I'm not sure this is a good assumption. Imagine adding a new finemapping method that requires a different criteria for QC. Then you need to add those flags to this class. It somehow doesn't feel general enough.